target_chains/ethereum/contracts/forge-test/PulseScheduler.t.sol

```diff
@@ -1092,6 +1100,102 @@ contract SchedulerTest is Test, SchedulerEvents, PulseSchedulerTestUtils {
         scheduler.updatePriceFeeds(subscriptionId, updateData, priceIds);
     }
 
+    function testUpdateSubscriptionEnforcesMinimumBalanceOnAddingFeeds()
+        public
+    {
+        // Setup: Create subscription with 2 feeds, funded exactly to minimum
+        uint8 initialNumFeeds = 2;
+        uint256 subscriptionId = addTestSubscriptionWithFeeds(
+            scheduler,
+            initialNumFeeds,
+            address(reader)
+        );
+        (
+            SchedulerState.SubscriptionParams memory currentParams,
+            SchedulerState.SubscriptionStatus memory initialStatus
+        ) = scheduler.getSubscription(subscriptionId);
+        uint256 initialMinimumBalance = scheduler.getMinimumBalance(
+            initialNumFeeds
+        );
+        assertEq(
+            initialStatus.balanceInWei,
+            initialMinimumBalance,
+            "Initial balance should be the minimum"
+        );
+
+        // Prepare new params with more feeds (4)
+        uint8 newNumFeeds = 4;
+        SchedulerState.SubscriptionParams memory newParams = currentParams;
+        newParams.priceIds = createPriceIds(newNumFeeds); // Increase feeds
+        newParams.isActive = true; // Keep it active
+
+        // Action 1: Try to update with insufficient funds
+        vm.expectRevert(abi.encodeWithSelector(InsufficientBalance.selector));
+        scheduler.updateSubscription(subscriptionId, newParams);
+
+        // Action 2: Add enough funds to meet the new minimum balance
+        uint256 newMinimumBalance = scheduler.getMinimumBalance(newNumFeeds);
+        uint256 requiredFunds = newMinimumBalance - initialMinimumBalance;
+        scheduler.addFunds{value: requiredFunds}(subscriptionId);
+
+        // Verification 2: Update should now succeed
+        vm.expectEmit();
+        emit SubscriptionUpdated(subscriptionId);
+        scheduler.updateSubscription(subscriptionId, newParams);
```

Comment:
I recommend you let updateSubscription be payable as well so people can update it in 1 go (now they need to add funds first, and then update it)